src/test/java/uk/gov/register/resources/RegisterCommandReaderTest.java

```diff
@@ -1,34 +1,32 @@
 package uk.gov.register.resources;
 
-import org.apache.commons.lang3.NotImplementedException;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
-import uk.gov.register.exceptions.SerializedRegisterParseException;
+import uk.gov.register.serialization.CommandParser;
 import uk.gov.register.serialization.RegisterCommand;
 import uk.gov.register.serialization.RegisterCommandList;
+import uk.gov.register.util.CanonicalJsonMapper;
+import uk.gov.register.util.CanonicalJsonValidator;
 
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.MultivaluedMap;
-import java.io.ByteArrayInputStream;
 import java.io.InputStream;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Type;
-import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Paths;
-import java.time.format.DateTimeParseException;
 import java.util.List;
 
-import static org.hamcrest.CoreMatchers.hasItems;
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
 
 @RunWith(MockitoJUnitRunner.class)
 public class RegisterCommandReaderTest {
-
     Class<RegisterCommandList> type = RegisterCommandList.class;
+    private RegisterCommandReader parser;
```

Comment:
Should this be renamed to `registerCommandReader` or something like `rsfParser`? Only because it looks a bit confusing in the constructor to have a variable called `commandParser` and a variable called `parser`.